DiscordChatExporter.Domain/Exporting/MediaDownloader.cs

```diff
@@ -50,14 +60,27 @@ public async ValueTask<string> DownloadAsync(string url)
 
     internal partial class MediaDownloader
     {
+        private static string HashUrl(string url)
+        {
+            // Knuth hash
+            UInt64 hashedValue = 3074457345618258791ul;
+            for (int i = 0; i < url.Length; i++)
+            {
+                hashedValue += url[i];
+                hashedValue *= 3074457345618258799ul;
+            }
+            return hashedValue.ToString();
+        }
+
```

Comment:
Any reason you chose this hash implementation? Just curious.

Reply:
This is a great question for which I do not have a good answer. I just Google'd something like "fast string hashing C#" and found this on some SO post. I completely forgot that `String.GetHashCode` is a thing in C#. I imagine that would suffice.

Reply:
Nevermind--I just remembered that `String.GetHashCode` is meant for hashtables and isn't [necessarily ideal for avoiding collisions](https://stackoverflow.com/a/7969189/8412850). Then again, the strings would be pretty long and there wouldn't be that many of them.

Reply:
But, then again, I also just remembered that we are also incorporating the original filenames into the filename we save, so even if a collision happened--it probably wouldn't matter.

Reply:
Honestly, I am not super sure what to do. The chance of the media filename and the url hash colliding seems absurdly low, but it's possible. My instinct is to play it safe with a cryptographic hash like SHA-256. We are hashing hundreds-of-thousands of strings, so I highly doubt it would result in any real performance impact.

Reply:
`String.GetHashCode` also returns different values for the same string across sessions/runtimes, so it's not useful to us. There are other hashes like md5/sha1 which could also be useful. Pretty sure they are CPU-optimized anyway so I doubt performance is a concern. I guess the important thing here is making sure we can actually cut off everything but the first 5 characters and still retain good entropy.

Reply:
`String.GetHashCode also returns different values for the same string across sessions/runtimes, so it's not useful to us.` Oh, that is good to know! 

From a couple of different SO posts [like this one](https://stackoverflow.com/a/18465323/8412850), truncating md5(or sha1) seems good enough.